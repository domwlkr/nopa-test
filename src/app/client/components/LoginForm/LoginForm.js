import React, {PropTypes} from 'react';
import { TextInput } from '../';

const LoginForm = (props) => {
  return (
    <form onSubmit={props.handleSubmit}>
      <TextInput
        name="surname"
        label="Surname"
        handleChange={props.handleChange}
        errors={props.errors.surname} />

      <TextInput
        name="sortCode"
        label="Sort code"
        handleChange={props.handleChange}
        errors={props.errors.sortCode} />

      <TextInput
        name="accountNumber"
        label="Account number"
        handleChange={props.handleChange}
        errors={props.errors.accountNumber} />

      <TextInput
        name="passCode"
        label="Passode"
        handleChange={props.handleChange}
        errors={props.errors.passCode}
        password />

      <TextInput
        name="memorableWord"
        label="Memorable word"
        handleChange={props.handleChange}
        errors={props.errors.memorableWord} />

      <input className="button" type="submit" value="Login &amp; connect" />
    </form>
  );
};

LoginForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  errors: PropTypes.object
};

export default LoginForm;
