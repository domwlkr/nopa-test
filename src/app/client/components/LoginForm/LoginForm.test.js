import React from 'react';
import LoginForm from './LoginForm';
import renderer from 'react-test-renderer';

const handleSubmit = jest.fn();
const handleChange = jest.fn();
const errors = {};

test('Renders correctly', () => {
  const loginForm = renderer.create(
    <LoginForm handleSubmit={handleSubmit} handleChange={handleChange} errors={errors} />
  ).toJSON();

  expect(loginForm).toMatchSnapshot();
});