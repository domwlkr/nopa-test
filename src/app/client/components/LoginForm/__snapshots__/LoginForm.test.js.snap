// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-input"
  >
    <label
      htmlFor="surname"
    >
      Surname
    </label>
    <input
      data-error={undefined}
      id="surname"
      name="surname"
      onChange={[Function]}
      placeholder="Surname"
      type="text"
    />
    <div
      className="errors"
    />
  </div>
  <div
    className="form-input"
  >
    <label
      htmlFor="sortCode"
    >
      Sort code
    </label>
    <input
      data-error={undefined}
      id="sortCode"
      name="sortCode"
      onChange={[Function]}
      placeholder="Sort code"
      type="text"
    />
    <div
      className="errors"
    />
  </div>
  <div
    className="form-input"
  >
    <label
      htmlFor="accountNumber"
    >
      Account number
    </label>
    <input
      data-error={undefined}
      id="accountNumber"
      name="accountNumber"
      onChange={[Function]}
      placeholder="Account number"
      type="text"
    />
    <div
      className="errors"
    />
  </div>
  <div
    className="form-input"
  >
    <label
      htmlFor="passCode"
    >
      Passode
    </label>
    <input
      data-error={undefined}
      id="passCode"
      name="passCode"
      onChange={[Function]}
      placeholder="Passode"
      type="text"
    />
    <div
      className="errors"
    />
  </div>
  <div
    className="form-input"
  >
    <label
      htmlFor="memorableWord"
    >
      Memorable word
    </label>
    <input
      data-error={undefined}
      id="memorableWord"
      name="memorableWord"
      onChange={[Function]}
      placeholder="Memorable word"
      type="text"
    />
    <div
      className="errors"
    />
  </div>
  <input
    className="button"
    type="submit"
    value="Login & connect"
  />
</form>
`;
